services/web-app/utils/use-query-state.js

```diff
@@ -57,15 +28,35 @@ export const useQueryState = (
       return null
     }
 
-    const query = new URLSearchParams(window.location.search)
-    const queryValue = query.get(key)
-    const storageValue = saveToStorage ? localStorage.getItem(`${router.pathname}:${key}`) : null
+    const query = queryString.parseUrl(window.location.search, {
+      ...queryStringConfig,
+      parseNumbers,
+      parseBooleans
+    }).query
+
+    const storageValue = saveToStorage
+      ? localStorage.getItem(`${router.pathname}:${key}`)
+      : undefined
+
+    let val = query[key]
 
-    return queryValue !== null ? parse(queryValue) : storageValue
-  }, [key, parse, router.pathname, saveToStorage])
+    if (val !== undefined && validateValue && !validateValue(val)) {
+      val = undefined
+      delete query[key]
 
-  // Update the "state" when the router.query key changes
-  const value = useMemo(getValue, [getValue, router.query[key]])
+      // Change query state without rerendering page
+      history.replaceState(null, null, `?${queryString.stringify(query, queryStringConfig)}`)
+    }
+
+    if (val === undefined) {
+      val = storageValue
+    }
+
+    return val
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Which deps does it want you to include that aren't currently in here?

Reply:
@jdharvey-ibm as a general practice, should we add comments next to any `// eslint-disable-next-line react-hooks/exhaustive-deps` to explain why we're disabling `react-hooks/exhaustive-deps`?

Reply:
Yeah! That'd be awesome!

Reply:
Agreed! will do

Reply:
Didn't need any extra dependencies for this case, removed comment